src/signal.rs

```diff
@@ -499,4 +505,89 @@ mod tests {
         // forever as the loop keeps running. Since we don't join, the thread will become detached
         // as the handle is dropped, and will be killed when the process/main thread exits.
     }
+
+    #[test]
+    fn test_get_blocked_signals() {
+        let mut sigset: sigset_t = unsafe { mem::zeroed() };
+
+        // Block a signal.
+        unsafe {
+            sigemptyset(&mut sigset);
+            sigaddset(&mut sigset, SIGRTMIN() + 0);
+            pthread_sigmask(SIG_BLOCK, &sigset, null_mut());
+        }
+
+        assert!(get_blocked_signals().unwrap().contains(&(SIGRTMIN() + 0)));
+    }
+
+    #[test]
+    fn test_block_signal() {
+        // Check if `SIGRTMIN() + 0` is blocked.
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, SIGRTMIN() + 0), 0);
+        }
+
+        block_signal(SIGRTMIN() + 0).unwrap();
+
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, SIGRTMIN() + 0), 1);
+        }
+    }
+
+    #[test]
+    fn test_unblock_signal() {
+        // Block a signal.
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            sigemptyset(&mut sigset);
+            sigaddset(&mut sigset, SIGRTMIN() + 1);
+            pthread_sigmask(SIG_BLOCK, &sigset, null_mut());
+        }
+        assert!(get_blocked_signals().unwrap().contains(&(SIGRTMIN() + 1)));
+
+        unblock_signal(SIGRTMIN() + 1).unwrap();
+
+        assert!(get_blocked_signals().unwrap().is_empty());
+    }
+
+    #[test]
+    fn test_clear_pending() {
+        unsafe {
+            register_signal_handler(
+                1,
+                SignalHandler::Siginfo(handle_signal_func),
+                true,
+                SA_SIGINFO,
+            )
+            .expect("failed to register vcpu signal handler");
+        }
+
+        // Block the signal, which means it won't be delivered until it is
+        // unblocked. Pending between the time when the signal which is set as blocked
+        // is generated and when is delivered.
+        let killable = thread::spawn(|| {
+            block_signal(SIGRTMIN() + 1).unwrap();
+        });
+
+        // Send a signal to the thread.
+        assert!(killable.kill(1).is_ok());
+
+        // Pending now.
+        unsafe {
+            assert!(!SIGNAL_HANDLER_CALLED_FLAG);
+        }
+
+        clear_signal(1).unwrap();
+
+        loop {
```

Comment:
Looks like this loop can get stuck if the flag never gets set. Does it make sense to simply wait some time period (like the current 100 ms) once, and then fail the test if the flag still didn't get set?

Reply:
Yes, so it means the clear_signal also fails when time is over after several loop. I'll update this later.